iR2_Alg MP.jl

```diff
@@ -15,12 +80,72 @@ mutable struct iR2Solver{R<:Real, S<:Vector} <: AbstractOptimizationSolver
   Fobj_hist::Vector{R}
   Hobj_hist::Vector{R}
   Complex_hist::Vector{Int}
-  p_hist::Vector{Vector{Int64}}
+  p_hist::Vector{Vector{Int}}
   special_counters::Dict{Symbol, Vector{Int}}
   h
   ψ
 end
 
+
+"""
+mutable struct iR2RegParams
+
+    pf::Int
+      current precision level of the smooth term
+
+    pg::Int
+      current precision level of the gradient of the smooth term
+
+    ph::Int
+      current precision level of the nonsmooth term
+
+    ps::Int
+      current precision level of the step
+
+    Π::Vector{DataType}
+      Contains the floating point formats used in the algorithm. For example, if Π = [Float16, Float32, Float64], then the algorithm will use Float16, Float32 and Float64.
+
+    verb::Bool
+      whether to display general information about the algorithm
+
+    activate_mp::Bool
+      whether to activate the mixed precision mode. If true, the algorithm will test the conditions of the mixed precision mode.
+
+    flags::Vector{Bool}
+      Contains some flags of the conditions of the mixed precision mode. If a flag is true, it means that the maximum precision level has been reached for the corresponding variable among : 
+        - f
+        - h
+        - ∇f
+        - s
+
+    κf::Real
```

Comment:
Not sure it's a good idea to have `Real` type for the parameters. It might causes type instability for the quantities computed with them.
You can opt  for MultiPrecisionR2-like implementation: choose a type parameter `H` as "high precision format" in which inexpensive computations are performed (`rho`, upper bounds on evaluation errors, ...).